zingolib/src/lightclient.rs

```diff
@@ -148,6 +150,69 @@ struct ZingoSaveBuffer {
     pub buffer: Arc<RwLock<Vec<u8>>>,
 }
 
+/// Balances that may be presented to a user in a wallet app.
+/// The goal is to present a user-friendly and useful view of what the user has or can soon expect
+/// *without* requiring the user to understand the details of the Zcash protocol.
+///
+/// Showing all these balances all the time may overwhelm the user with information.
+/// A simpler view may present an overall balance as:
+///
+/// Name | Value
+/// --- | ---
+/// "Balance" | `spendable` - `minimum_fees` + `immature_change` + `immature_income`
+/// "Incoming" | `incoming`
+///
+/// If dust is sent to the wallet, the simpler view's Incoming balance would include it,
+/// only for it to evaporate when confirmed.
+/// But incoming can always evaporate (e.g. a transaction expires before confirmation),
+/// and the alternatives being to either hide that a transmission was made at all, or to include
+/// the dust in other balances could be more misleading.
+///
+/// An app *could* choose to prominently warn the user if a significant proportion of the incoming balance is dust,
+/// although this event seems very unlikely since it will cost the sender *more* than the amount the recipient is expecting
+/// to 'fool' them into thinking they are receiving value.
+/// The more likely scenario is that the sender is trying to send a small amount of value as a new user and doesn't realize
+/// the value is too small to be useful.
+/// A good Zcash wallet should prevent sending dust in the first place.
+pub struct UserBalances {
+    /// Available for immediate spending.
+    /// Expected fees are *not* deducted from this value, but the app may do so by subtracting `minimum_fees`.
+    /// `fairy_dust` is excluded from this value.
+    ///
+    /// For enhanced privacy, the minimum number of required confirmations to spend a note is usually greater than one.
+    pub spendable: u64,
+
+    /// The sum of the change notes that have insufficient confirmations to be spent.
+    pub immature_change: u64,
+
+    /// The minimum fees that can be expected to spend all `spendable + immature_change` funds in the wallet.
+    /// This fee assumes all funds will be sent to a single note.
+    ///
+    /// Balances described by other fields in this struct are not included because they are not confirmed,
+    /// they may amount to dust, or because as `immature_income` funds they may require shielding which has a cost
+    /// and can change the amount of fees required to spend them (e.g. 3 UTXOs shielded together become only 1 note).
+    pub minimum_fees: u64,
+
+    /// The sum of non-change notes with a non-zero confirmation count that is less than the minimum required for spending,
+    /// and all UTXOs (considering that UTXOs must be shielded before spending).
```

Comment:
This policy makes perfect sense for the wallet library I'm building on zingolib, because I'm going to apply auto-shielding. But last I heard from the zingo team, zingolib wasn't going that route (nor were the zingo wallet apps). Is that still the case? If so, what do you want to do about the way this struct describes the balances?

We could potentially break out confirmed transparent income from this field to track it separately.

Reply:
Breaking out confirmed transparent balance to display separately is probably the way to go here.

Reply:
That calls into question whether `minimum_fees` includes UTXOs (which it didn't before), and makes it overall more confusing because `spendable` no longer represents all spendable funds, once you consider UTXOs to be spendable.
I'm going to investigate the option of just taking a parameter to indicate how to treat transparent funds, because given one or the other policy, I think we can fit into this struct in a consistent way, although the docs will get more complicated as it has to describe either case.